xaod_cpp_transformer/transformer.py

```diff
@@ -51,8 +51,64 @@
 object_store = None
 posix_path = None
 
+instance = os.environ.get('INSTANCE_NAME', 'Unknown')
 
-def initialize_logging(request=None):
+
+# class StreamFormatter(logging.Formatter):
+#     """
+#     A custom formatter that adds extras.
+#     Normally log messages are "level instance component msg extra: {}"
+#     """
+#     def_keys = ['name', 'msg', 'args', 'levelname', 'levelno',
+#                 'pathname', 'filename', 'module', 'exc_info',
+#                 'exc_text', 'stack_info', 'lineno', 'funcName',
+#                 'created', 'msecs', 'relativeCreated', 'thread',
+#                 'threadName', 'processName', 'process', 'message']
+
+#     def format(self, record):
+#         """
+#         :param record: LogRecord
+#         :return: formatted log message
+#         """
+
+#         string = super().format(record)
+#         extra = {k: v for k, v in record.__dict__.items()
+#                  if k not in self.def_keys}
+#         if len(extra) > 0:
+#             string += " extra: " + str(extra)
+#         return string
+
+
+class LogstashFormatter(logstash.formatter.LogstashFormatterBase):
```

Comment:
Can this not be in the transformer library? Is it specific to the xaod transformer?

Reply:
It could. There is nothing specific to xaod transformer.  But in transformer library:
```
    def __init__(self, servicex_endpoint, logger=None):
        # Default logger doesn't print so that code that uses library
        # can override
```
Later we can think how to do this centrally for all the services.